sympy/matrices/matrices.py

```diff
@@ -1141,15 +1164,25 @@ def eigenvals(self, error_when_incomplete=True, **flags):
                     eigs[diagonal_entry] += 1
         else:
             flags.pop('simplify', None)  # pop unsupported flag
-            eigs = roots(mat.charpoly(x=Dummy('x')), **flags)
+            eigs = roots(mat.charpoly(x=Dummy('x'), simplify=simpfunc_1), **flags)
```

Comment:
Perhaps `if isinstance(simplify, FunctionType):` could be here. Then `charpoly` could be called with `simplify=simplify`. Otherwise the line above would be used.